### PR TITLE
Add templateId to CreateScheduledWebinarAsync

### PR DESCRIPTION
**Pull request by MrSimonC**
OK, so this is my **FIRST ever pull request to a project which isn't mine** - so be kind! eeep!

I've just started on a new project which uses your excellent nuget package, however, my client has set up a templates in Zoom and wishes to use a template when the Zoom webinar is created via the Zoom API.

My research has lead me to this:
https://devforum.zoom.us/t/create-webinar-through-api-using-webinar-template/33013/4
... aka the templateId should be passed to the API as a parameter:
`https://api.zoom.us/v2/users/{userId}/webinars?template_id={templateId}`
... which the Zoom team haven't put in the official documentation.

There is, however, an existing "[List Webinar Templates](https://marketplace.zoom.us/docs/api-reference/zoom-api/webinars/listwebinartemplates)" end point which provides the templateIds (but again, only that above link exists in how to use them).

Can you consider my Pull Request please?
My only other point I should make is that I did look at the tests solution, but since it's an integration test, the test's templateId will need to exist before it's used in the test runs - so I didn't update the integration test due to this.

### CODE REVIEW

**Comment by Jericho** [comment]
> OK, so this is my **FIRST ever pull request to a project which isn't mine** - 

Awesome! Welcome to the community. I sincerely hope this library will be useful to you and I am grateful that you are taking the time to contribute.

> so be kind! eeep!

What are you talking about? I'm always kind, of course!!! cue evil laugh ... Mwahahaha!!! :grin: :stuck_out_tongue_closed_eyes: :smiling_imp:

Joking aside, I definitely want to foster a welcoming environment where contributors want to participate and contribute back so I'll do my best to guide you in the most positive and constructive way I can.

The first feedback I would like to share is that it's always preferable to start by raising an issue when you want to contribute to a project. The issue allows you to clearly explain the problem you are facing, describe the changes you intend to make, share the ideas you have, ask questions, etc. and it allows the maintainer and other contributors to engage in a discussion with you. In the course of the discussion, the maintainer may give you pointers on how they would like the problem to be resolved, someone may have already faced this situation and may have a workaround, etc. It would be a shame for you to invest time and effort in writing code only to find out upon submitting your PR that there is a already a solution/workaround or that the maintainer is not interested in the solution you are offering. There is another reason for raising issues but this one is specific to the projects that I maintain: I use a tool that automatically generates release notes whenever I publish a new version and this tool uses the title of the issues to populate the release notes (you can see release notes [here](https://github.com/Jericho/ZoomNet/releases)). Therefore it's crucial to have an issue for each PR in order for the change/fix/new feature to be mentioned in the release notes.

> the templateId should be passed to the API as a parameter:

This is news to me! I didn't know about this so thanks for sharing this knowledge with me.

> ... which the Zoom team haven't put in the official documentation.

Indeed. When I first read your comment I was afraid I might have missed something in the documentation but I just double-checked and there is no mention of the templateId anywhere.

> There is, however, an existing "[List Webinar Templates](https://marketplace.zoom.us/docs/api-reference/zoom-api/webinars/listwebinartemplates)" end point which provides the templateIds (but again, only that above link exists in how to use them).

Retrieving the list of existing webinar templates is missing from ZoomNet and it's definitely worthy of a separate issue+PR so we add this functionality.

> Can you consider my Pull Request please?

Absolutely! I will review what you submitted and will utilize GitHub's review functionality to provide feedback on the changes you are submitting.

> My only other point I should make is that I did look at the tests solution, but since it's an integration test, the test's templateId will need to exist before it's used in the test runs - so I didn't update the integration test due to this.

Understood. Their API doesn't seem to provide a way to create/update/delete webinar templates, unfortunately.

File: Source/ZoomNet/Resources/IWebinars.cs
Comment (Jericho) [inline Source/ZoomNet/Resources/IWebinars.cs:59]: Generally speaking it's good practice to add new parameters at the end, as you have done, but there is one notable exception to this rule (at least in my opinion): the CancellationToken should always be the last parameter in an async method signature. So, please position the new templateId parameter before the cancellationToken.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (Jericho) [inline Source/ZoomNet/Resources/Webinars.cs:102]: See my previous comment regarding position of the templateId parameter.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (Jericho) [inline Source/ZoomNet/Resources/Webinars.cs:117]: This is unnecessary, please remove this line and see my next comment for a simpler way to append querystring parameters to a request.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (Jericho) [inline Source/ZoomNet/Resources/Webinars.cs:120]: Here's a simpler way to append querystring parameters:
```suggestion
				.PostAsync($"users/{userId}/webinars")
				.WithArgument("template_id", templateId)
```

**Review by Jericho — CHANGES_REQUESTED** [review]
Looks good. I made a few comments.

File: Source/ZoomNet/Resources/IWebinars.cs
Comment (MrSimonC) [reply Source/ZoomNet/Resources/IWebinars.cs]: Great suggestion - I also agree (and should have spotted this). I've updated the code.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (MrSimonC) [reply Source/ZoomNet/Resources/Webinars.cs]: Great suggestion - I also agree (and should have spotted this). I've updated the code.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (MrSimonC) [reply Source/ZoomNet/Resources/Webinars.cs]: Great suggestion. I've updated the code. I'm not used to the fluent httpclient you're using.

File: Source/ZoomNet/Resources/Webinars.cs
Comment (MrSimonC) [reply Source/ZoomNet/Resources/Webinars.cs]: Code line removed.

**Comment by MrSimonC** [comment]
Hello! Sorry for the late reply (Covid, childcare from home, UK lockdown etc etc)!
Many thanks indeed for the helpful text and guidance. I did consider creating an issue, but I thought "I don't want to be a bother" - but never considered it could be used for automated release notes! nice! I'll definitely follow that route in future.

I've made the code change recommendations you suggested. As I'm new, I've tried updating my code branch and pushing it up. I can see that github has labelled my code (above in this chat) as "outdated", but I do see that it's also registered my pushed up commit. I guess that's normal?!

Maybe next time i'll try that new functionality of applying your code suggestions straight as a commit (does that go into my repo though, since the PR comes from that? then i just pull down the github-suggestion-commit?). So many questions! Sorry!

I'll mark the above issues as resolved since I've made changes to the code. Can i ask - at work normally the person doing the code Resolves the comment - but is that right? i.e. should it instead be the reviewer who finally marks it as resolved?

I've also created 2 issues as you've suggested - one for this (and have referenced this PR) and one to add functionality to look up templateIds. My only question is - how do you know'll attempt the code change for the commit (i.e. how do you avoid two developers accidently doing the same code to fulfil the issue? Just work it out in discussion on the issue?

**Comment by Jericho** [comment]
> I've made the code change recommendations you suggested. As I'm new, I've tried updating my code branch and pushing it up. I can see that github has labelled my code (above in this chat) as "outdated", but I do see that it's also registered my pushed up commit. I guess that's normal?!

Yes indeed, totally normal. 

> Maybe next time i'll try that new functionality of applying your code suggestions straight as a commit (does that go into my repo though, since the PR comes from that? then i just pull down the github-suggestion-commit?).

Yes it does. It's a very handy feature that GitHub added not too long ago (I don't remember exactly when they added it, maybe a year ago or so). I personally love it. Sometimes it can take much longer to describe the desired change in plain English and you run the risk of you explanation to be misunderstood, which causes the contributor to submit a correction that doesn't match what you want, and therefore another round of review/correction and so forth. You end up wasting a lot of time.

> I'll mark the above issues as resolved since I've made changes to the code. Can i ask - at work normally the person doing the code Resolves the comment - but is that right? i.e. should it instead be the reviewer who finally marks it as resolved?

Yes that's exactly the process. You resolve each review comment which let's me know that you have addresses each of my comments. Ultimately the project admin (which in this case is me) is the one who approves the changes as a whole before the PR can be merged.

> I've also created 2 issues as you've suggested - one for this (and have referenced this PR) and one to add functionality to look up templateIds. 

Excellent. Thanks for doing this.

> My only question is - how do you know'll attempt the code change for the commit (i.e. how do you avoid two developers accidently doing the same code to fulfil the issue? Just work it out in discussion on the issue?

That's a good question, I don't know the perfect answer but a comment indicating that you are volunteering to resolve the issue is probably the best way. From my perspective, I can assign the issue to you to make it clear that I am delegating to you. Makes sense?

**Review by Jericho — APPROVED** [review]
Looks good to me!

Thank you for your contribution.

**Comment by MrSimonC** [comment]
Thanks for answering my questions - it's all starting to make sense now!
Last question, if it's all ok, _(I know only you have write access, and I don't - and that's ok)_ am I OK to just leave everything as it is and you'll deal with merging the branch in, when you're ready? (no rush on my end) (i.e. I *shouldn't* press "Close with comment" as that'll close the PR without merging?

**Comment by Jericho** [comment]
If you "close" the PR (with or without a comment), it basically mean that you are abandoning it and it tells me that I shouldn't merge it. In fact, I believe that GitHub won't even allow me to merge a closed PR and I would have to reopen it if I want to merge it.

At this time, you've addressed all my feedback so the ball is in my court to approve your changes (which I did a short while ago) and to merge the PR (which I'll do as soon as I'm done typing this comment).

Once again, thank you for your contribution. It's much appreciated!

**Pull request merged**
